compiler/AST/foralls.cpp

```diff
@@ -156,6 +156,8 @@ ShadowVarSymbol* ShadowVarSymbol::buildForPrefix(ShadowVarPrefix prefix,
                                     Expr* nameExp, Expr* type, Expr* init)
 {
   const char* nameString = toUnresolvedSymExpr(nameExp)->unresolved;
+  if (nameString == astrThis)
+    USR_FATAL_CONT(nameExp, "cannot apply a forall or task intent to 'this'");
```

Comment:
Can we change it to "cannot currently apply a forall or task intent to 'this'" since this seems like more of a PR giving a better error for the current state even though we probably expect one day to allow it?